File: MoveZeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        if(nums.length == 1){
            return;
        }
        
        int pointer = 0;
        
        for(int i=0; i<nums.length; i++){
            if(nums[i] != 0){
                nums[pointer] = nums[i];
                if(i != pointer){
                    nums[i] = 0;   
                }
                
                pointer++;
            }
        }
        
    }
}

File: MoveZeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let index = 0;
    nums.forEach((currentElement, i) => {
        if(currentElement !== 0) {
            if(index !== i) {
                nums[index] = currentElement;
                nums[i] = 0;
            }
        index++;
        }
    });
};
